src/submission/validation-clinical/donor.ts

```diff
@@ -42,11 +57,18 @@ export const validate = async (
   }
 
   // ***Submission Validation checks***
-  const errors: SubmissionValidationError[] = []; // all errors for record
+  let errors: SubmissionValidationError[] = []; // all errors for record
   // cross entity donor-specimen record validation
   checkTimeConflictWithSpecimens(submittedDonorClinicalRecord, mergedDonor, errors);
 
   // other checks here and add to `errors`
+  const crossFileErrors = await crossFileValidator(
+    submittedDonorClinicalRecord,
+    submittedRecords,
+    mergedDonor,
+  );
+
+  errors = [...errors, ...crossFileErrors];
 
   return { errors };
```

Comment:
A bit confused about these changes here. The purpose of this logic seems unclear at first glance, so let me try to walk through it as I reason with it like a new team member would in a few months/years:

Initially, the empty `errors` array was a `const`, but needed to be changed into a `let`, presumably so it can be mutated in line 71. This can be avoided by simply returning a new array inline (see suggestion below).

However, now I wonder what the point of `errors` array itself was to begin with, defined as an empty array; of course, unless it's needed by `checkTimeConflictWithSpecimens` who may be mutating it elsewhere... which is both not obvious here, and possibly a source of rather hard to debug issues later on (and then again I'm speculating what is happening here, as I didn't go looking deeper into that function).
The code as it was doesn't really explain what it's meant to do.

Looks to me (guessing) like leaving the `const` unmodified, and just giving a different return for the whole function would be a cleaner change here; but that still doesn't clarify to me why the return needed an empty array to begin with, unless that other function is indeed mutating it elsewhere which is smelly code.

```suggestion
  const errors: SubmissionValidationError[] = []; // all errors for record
  // cross entity donor-specimen record validation
  checkTimeConflictWithSpecimens(submittedDonorClinicalRecord, mergedDonor, errors);

  // other checks here and add to `errors`
  const crossFileErrors = await crossFileValidator(
    submittedDonorClinicalRecord,
    submittedRecords,
    mergedDonor,
  );

  return { errors: [...errors, ...crossFileErrors] };
```

Reply:
Yes, updating the return value is a great idea. 

So the back story here is a lot of the validation is using `.push` and it's across a bunch of functions, in this case `checkTimeConflictWithSpecimens`.

I really prefer keeping things functional/encapsulated so it's a mismatch of patterns btwn `crossFileErrors` and `checkTimeConflictWithSpecimens`

But now that I look at it, `checkTimeConflictWithSpecimens` is not exported or used anywhere else, so maybe there's a chance to refactor here.

Reply:
Oh, that's great to hear! I'll leave this in your hands then.
Thanks for hearing me out, Dan 🙏 

While Mutation is of course a valid and useful approach used by several companies and indy devs out there, as you may have found out in this app it tends to get messy over time. Therefor, I have a personal rule to avoid `let` altogether, and whenever I've found it being used, it is almost always a smell of brittle code.

Let me know when you've made whatever changes you see fit, and I'll gladly give this PR another look 💪

Reply:
Totally agree, felt wrong adding a `let` ;)